src/scaffoldmaker/meshtypes/meshtype_3d_lung4.py

```diff
@@ -0,0 +1,804 @@
+from scaffoldmaker.meshtypes.scaffold_base import Scaffold_base
+from scaffoldmaker.utils.geometry import getEllipsePointAtTrueAngle
+from scaffoldmaker.utils.meshrefinement import MeshRefinement
+from scaffoldmaker.utils.ellipsoidmesh import EllipsoidMesh, EllipsoidSurfaceD3Mode
+from scaffoldmaker.utils.zinc_utils import translate_nodeset_coordinates
+
+import copy
+import math
+
+
+class MeshType_3d_lung4(Scaffold_base):
+    """
+    Generates a lung scaffold with common hilum but open fissures with lobes built from 1/6 ellipsoid segments.
+    """
+
+    @classmethod
+    def getName(cls):
+        return "3D Lung 4"
+
+    @classmethod
+    def getParameterSetNames(cls):
+        return [
+            "Default",
+            "Human 1 Coarse",
+            "Human 1 Medium",
+            "Human 1 Fine",
+            "Ellipsoid"
+        ]
+
+    @classmethod
+    def getDefaultOptions(cls, parameterSetName="Default"):
+        options = {}
+        useParameterSetName = "Human 1 Coarse" if (parameterSetName == "Default") else parameterSetName
+        options["Left lung"] = True
+        options["Right lung"] = True
+        options["Ellipsoid height"] = 1.0
+        options["Ellipsoid dorsal-ventral size"] = 0.8
+        options["Ellipsoid medial-lateral size"] = 0.4
+        options["Left-right lung spacing"] = 0.5
+        options["Lower lobe extension"] = 0.3
+        options["Refine"] = False
+        options["Refine number of elements"] = 4
+
+        if "Medium" in useParameterSetName:
+            options["Number of elements lateral"] = 4
+            options["Number of elements lower lobe extension"] = 3
+            options["Number of elements oblique"] = 8
+            options["Number of transition elements"] = 1
+        elif "Fine" in useParameterSetName:
+            options["Number of elements lateral"] = 6
+            options["Number of elements lower lobe extension"] = 4
+            options["Number of elements oblique"] = 12
+            options["Number of transition elements"] = 1
+        else:
+            options["Number of elements lateral"] = 4
+            options["Number of elements lower lobe extension"] = 2
+            options["Number of elements oblique"] = 6
+            options["Number of transition elements"] = 1
+
+        if "Human" in useParameterSetName:
+            options["Lower lobe base concavity"] = 0.1
+            options["Ventral edge sharpness factor"] = 0.9
+            options["Cardiac curvature"] = 2.0
+        else:
+            options["Lower lobe base concavity"] = 0.0
+            options["Ventral edge sharpness factor"] = 0.0
+            options["Cardiac curvature"] = 0.0
+
+        return options
+
+    @classmethod
+    def getOrderedOptionNames(cls):
+        return [
+            "Left lung",
+            "Right lung",
+            "Number of elements lateral",
+            "Number of elements lower lobe extension",
+            "Number of elements oblique",
```

Comment:
Would it be clearer if you rename this as 'Number of elements along oblique fissure"?

Reply:
I think it's the best for now and the oblique plane is understood by lung people. It's not strictly true for the fissure because there are fewer on the lower lobe fissure (half + extension).
Also it may be possible to build the mesh with a different number horizontally, so this leaves that option open (vs. having number of elements radial (half the number) which is not good to mix with the number in the lateral direction (medial-lateral) which I don't want to use half the number for.